src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -0,0 +1,217 @@
+//   - This script is committed to ./src/System.Private.ServiceModel/tools/PRService
+//   - 
+//   - The Git repo is set up as follows: 
+//     [remote "origin"]
+//	       url = https://github.com/dotnet/wcf
+//	       fetch = +refs/heads/*:refs/remotes/origin/*
+//	       fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Web;
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files\Git\bin\git.exe";
+    private static string _gitRepoPath;
+    private const int _gitExecutionTimeoutMilliseconds = 100000;
+
+    public void ProcessRequest(HttpContext context)
+    {
+        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+        string prString = context.Request.QueryString["pr"];
+
+        context.Response.Write(string.Format("PR = {0}<br/>", HttpUtility.HtmlEncode(prString) ?? "Unspecified"));
+
+        StringBuilder result = new StringBuilder();
+
+        bool success = CheckPrerequisites(result);
+
+        if (success) 
+        {
+            success = CleanupMergedBranches(result); 
+        }
+
+        if (success)
+        {
+            // Check inputs to the script
+            uint pr;
+            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
+            {
+                success = SyncToPr(pr, result);
+            }
+            else
+            {
+                context.Response.Write(string.Format("Invalid PR ID specified<br/>"));
+                success = false;
+            }
+        } 
+        
+        if (!success)
+        {
+            context.Response.StatusCode = 400;
+        }
+
+        if (context.Request.IsLocal)
+        {
+            // Don't allow error results to leak to non-local machines
+            context.Response.Write(HttpUtility.HtmlEncode(result));
+        }
+    }
+
+    // Check prerequisites for whether or not we can run
+    private bool CheckPrerequisites(StringBuilder executionResult)
+    {
+        if (!File.Exists(_gitExecutablePath))
+        {
+            executionResult.AppendFormat("Could not find or insufficient permissions to access the git executable at path '{0}'", _gitExecutablePath);
+            return false;
+        }
+
+        if (!Directory.Exists(_gitRepoPath))
+        {
+            executionResult.AppendFormat("Could not find or insufficient permissions to access the git repo at path '{0}'", _gitRepoPath);
+            return false;
+        }
+
+        // Do a sanity execute of the git executable 
+
+        try
+        {
+            string[] gitCommands = new string[]
+            {
+                "status"
+            };
+
+            RunGitCommands(gitCommands, executionResult);
```

Comment:
Nit: check success return from the 'git status'?  If that fails, I don't believe we will see an exception and CheckPrerequisites will return true.